modules/apps/translation/translation-test/src/testIntegration/java/com/liferay/translation/info/item/updater/test/InfoFormValuesUpdaterTest.java

```diff
@@ -157,12 +230,131 @@ public void testUpdateArticleFromInfoFormValuesOnlyTitle()
+		).put(
+			_LOCALE_US, RandomTestUtil.randomString()
+		).build();
+
+		Map<Locale, String> descriptionMap = HashMapBuilder.put(
+			_LOCALE_JA, "translate description to japanese"
+		).put(
+			_LOCALE_US, RandomTestUtil.randomString()
+		).build();
+
+		Map<Locale, String> titleMap = HashMapBuilder.put(
+			_LOCALE_JA, "translate title to japanese"
+		).put(
+			_LOCALE_US, RandomTestUtil.randomString()
+		).build();
+
+		JournalArticle article = JournalTestUtil.addArticle(
+			_group.getGroupId(), 0,
+			PortalUtil.getClassNameId(JournalArticle.class), titleMap,
+			descriptionMap, contentMap, LocaleUtil.getSiteDefault(), false,
+			true, _serviceContext);
+
+		Assert.assertEquals(
+			"translate title to japanese", article.getTitle(_LOCALE_JA));
+
+		Assert.assertEquals(
+			"translate description to japanese",
+			article.getDescription(_LOCALE_JA));
+
+		Assert.assertEquals(
+			"translate content to japanese",
+			_assertTranslatedContent(
+				article.getContent(), "name", _LOCALE_US, _LOCALE_JA));
+
+		InfoFormValues infoFormValues =
+			_xliffTranslationInfoFormValuesImporter.importXLIFF(
+				_group.getGroupId(),
+				new InfoItemClassPKReference(
+					JournalArticle.class.getName(), 122),
+				TranslationTestUtil.readFileToInputStream(
+					"test-journal-article_122_Jap.xlf"));
+
+		JournalArticle journalArticle =
+			_infoFormValuesUpdater.updateFromInfoFormValues(
+				article, infoFormValues);
+
+		Assert.assertEquals("これはタイトルです", journalArticle.getTitle(_LOCALE_JA));
+
+		Assert.assertEquals(
+			"これは要約です", journalArticle.getDescription(_LOCALE_JA));
+
+		Assert.assertEquals(
+			"<p>これが内容です</p>",
+			_assertTranslatedContent(
+				journalArticle.getContent(), "name", _LOCALE_US, _LOCALE_JA));
+	}
+
+	private String _assertTranslatedContent(
+			String actualXML, String fieldName, Locale sourceLocale,
+			Locale targetLocale)
+		throws Exception {
+
+		Document actualDocument = SAXReaderUtil.read(actualXML);
+
+		Element rootElement = actualDocument.getRootElement();
+
+		String availableLanguageIds = rootElement.attributeValue(
+			"available-locales");
+
+		if (!availableLanguageIds.contains(sourceLocale.toString()) ||
+			!availableLanguageIds.contains(targetLocale.toString())) {
+
+			return StringPool.BLANK;
+		}
+
+		for (Element dynamicElementElement :
+				rootElement.elements("dynamic-element")) {
+
+			String attribute = dynamicElementElement.attributeValue(
+				"name", StringPool.BLANK);
+
+			if (!Objects.equals(attribute, fieldName)) {
+				return StringPool.BLANK;
+			}
+
+			for (Element element :
+					dynamicElementElement.elements("dynamic-content")) {
+
+				String languageId = element.attributeValue(
+					"language-id", StringPool.BLANK);
+
+				if (Objects.equals(
+						languageId, LocaleUtil.toLanguageId(targetLocale))) {
+
+					return element.getStringValue();
+				}
+			}
+		}
+
+		return StringPool.BLANK;
+	}
+
+	private static final Locale _LOCALE_ES = LocaleUtil.SPAIN;
```

Comment:
Hi @AliciaGarciaGarcia,

Why do we need to extract these as constants? They are already constants and we don't do this in any other place.

I would use them explicitly without extracting them to constants. 

Also, if you really want to do it, it would be better to do it in a separate commit that is easily revertible if Brian doesn't like it.